src/lib/storage/value_segment.cpp

```diff
@@ -150,12 +156,16 @@ template <typename T>
 std::shared_ptr<BaseSegment> ValueSegment<T>::copy_using_allocator(const PolymorphicAllocator<size_t>& alloc) const {
   pmr_concurrent_vector<T> new_values(_values, alloc);  // NOLINT(cppcoreguidelines-slicing)
                                                         // (clang-tidy reports slicing that comes from tbb)
+
+  std::shared_ptr<BaseSegment> copied_segment;
   if (is_nullable()) {
     pmr_concurrent_vector<bool> new_null_values(*_null_values, alloc);  // NOLINT(cppcoreguidelines-slicing) (see above)
-    return std::allocate_shared<ValueSegment<T>>(alloc, std::move(new_values), std::move(new_null_values));
+    copied_segment = std::allocate_shared<ValueSegment<T>>(alloc, std::move(new_values), std::move(new_null_values));
   } else {
-    return std::allocate_shared<ValueSegment<T>>(alloc, std::move(new_values));
+    copied_segment = std::allocate_shared<ValueSegment<T>>(alloc, std::move(new_values));
   }
+  copied_segment->access_counter.set_counter_values(access_counter);
```

Comment:
We should discuss what the expected behavior here is. I could find reasons for both the values to be copied or to be reset.

Reply:
That is a tough question.

1. A freshly copied segment hasn’t been accessed yet. One could expect the counters to be zero.
2. When copying an object, one might expect an exact duplicate.

Personally, I think I would expect an exact duplicate. I can always reset the counters manually, if required, and will know, that I am not holding an exact copy anymore.

Reply:
2 sounds reasonable. That should explicitly be documented somewhere.